lib/paypal_express/api.rb

```diff
@@ -59,15 +60,27 @@ def purchase_payment(kb_account_id, kb_payment_id, kb_payment_transaction_id, kb
         # Pass extra parameters for the gateway here
         options = {}
 
-        add_required_options(kb_payment_transaction_id, kb_payment_method_id, context, options)
+        if find_value_from_properties(properties, 'from_hpp')
```

Comment:
The `from_hpp` property won't be specified by the caller. Instead, we should just check if we have a `baid` on file or not.